src/rantanplan/core.py

```diff
@@ -99,69 +99,96 @@ def have_prosodic_liaison(first_syllable, second_syllable):
             and second_syllable['syllable'][0] in LIAISON_SECOND_PART)
 
 
-def get_synalephas(words):
+def get_synalephas(sinaereris_words_dictionary):
+    """
+    Gets a list of dictionaries for each word on each line of the poem
+    and joins the syllables to create phonologic syllables according to its
+    'has_synaloepha' value
+    :param sinaereris_words_dictionary: List of dictionaries for each word of the poem
+    :return: A list of conjoined syllables
+    """
     prosodic_line = []
-    word_list = copy.deepcopy(words)
+    word_list = copy.deepcopy(sinaereris_words_dictionary)  # create a deepcopy since we're going to pop()
     for idx, syllables_list in enumerate(word_list):
         for syllable in syllables_list:
             if not syllable.get('has_synalepha'):
                 prosodic_line.append(syllable)
             else:
                 next_syllable = word_list[idx + 1].pop(0)
```

Comment:
I still have some concerns with this implementation. A `for...loop` with a `.pop()` on the control variable.

Reply:
It works for now. Maybe we can handle it later.
For what I recall it was the easiest way to solve an extra syllable problem. It also has to do with changing the list elements of the assigned variable that affects the main variable and I didn't want that.

Reply:
That all can be fixed with indices, which avoids the duplicated data.